### PR TITLE
Implement 'GET /portfolioDrafts' 

### PR DESCRIPTION
**Pull request by jasonburkert-ccpo**
Adds the capability to retrieve a collection of portfolio drafts.
Implements `GET /portfolioDrafts`

Ticket: AT-6362

### CODE REVIEW

**Comment by jasonburkert-ccpo** [comment]
Output may need some adjustment.

image

File: provisioning/src/handlers/portfolioDrafts/get.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  const connectionString = process.env['COSMOS-CONNECTION-STRING']
+  context.log('Connecting to ' + connectionString)
+  const client = new CosmosClient(connectionString)
+
+  // TODO: create these in terraform & fail fast
+  const { database } = await client.databases.createIfNotExists({ id: 'atat' })
+  const { container } = await database.containers.createIfNotExists({ id: 'portfolios' })
+
+  // TODO: add a modicum of error handling
+  const results = await container.items.readAll().fetchAll()
```
Comment (jeffsegal-ccpo) [inline provisioning/src/handlers/portfolioDrafts/get.ts:14]: We need to a slight transformation here to that the response matches the spec:

```
[
  {
      "id": "283c12eb-377f-4df5-82d0-5e553bab445c",
      "created_at": "{2021} 07-22T08:57:09 068 Z AM",
      "updated_at": "{2021} 07-22T08:57:09 068 Z AM",
      "status": "not_started"
  }
]
```

Right now it's wrapped by a `resources` property and also includes the cosmos metadata that we don't want to expose.

**Review by jeffsegal-ccpo — CHANGES_REQUESTED** [review]
We absolutely need unit tests but I'm fine with punting since we don't have much of any conditional logic in here yet. We'll get a few tickets put together to address that, validation and error handling.

File: provisioning/src/handlers/portfolioDrafts/get.ts (hunk 1 above)
Comment (jasonburkert-ccpo) [reply provisioning/src/handlers/portfolioDrafts/get.ts:14]: That's exactly the type feedback I was seeking.  Thanks.

Comment (zachclark-ccpo) [reply provisioning/src/handlers/portfolioDrafts/get.ts:14]: A for loop to map each result to the ProvisioningStatus.ts model would be a good way to implement this.

Comment (anhtran-ccpo) [reply provisioning/src/handlers/portfolioDrafts/get.ts:14]: To expand on what zach is saying 

```
const results = await container.items.readAll().fetchAll().map((data) => {
      data.id,
      data.created_at
      data.updated_at,
      data.status
     })
```

Comment (jeffsegal-ccpo) [reply provisioning/src/handlers/portfolioDrafts/get.ts:14]: I miss the Java Streaming API....  :'(

Comment (jasonburkert-ccpo) [reply provisioning/src/handlers/portfolioDrafts/get.ts:14]: Very helpful. Thank you.

**Comment by jasonburkert-ccpo** [comment]
Closing due file relocation from #24.  Will open new PR.

**Pull request closed without merging**